### ISSUE TITLE
got different seedhash between eth_getwork and ethash cache generate

### ISSUE THREAD

**Issue by iamleoleung** [etclabscore/core-geth#238]
#### System information

Geth version: `1.11.18-unstable`
OS & Version: Linux
Commit hash : e89f93ef7d9dd7b45f815b62bc235d53b406793d

#### Expected behaviour

I got a different seedhash when i run mordor

current block number: 2698351
when i eth_getWork(), got seedhash is:  0518e1373ff88ccabb28493cac10cb0731313135d880dae0d846be6016ab9acf

but when verify,  seedhash got: 9336adfd9667a72451ef339e888c4053d10c735d612950f05fbe22d9a139e54c

I think the difference is because

in eth_getWork(), calculate seedhash via block number
```
line 348 in consensus/ethash/sealer.go
s.currentWork[1] = common.BytesToHash(SeedHash(block.NumberU64())).Hex()
```

but i cache seedhash
```
line 305 in consensus/ethash/ethash.go 
seed := seedHash(c.epoch*c.epochLength + 1)
```

both will call 
```
line 142 in. core-geth/consensus/ethash/algorithm.go
   135	// seedHash is the seed to use for generating a verification cache and the mining
   136	// dataset.
   137	func seedHash(block uint64) []byte {
   138		seed := make([]byte, 32)
   139		if block < epochLengthDefault {
   140			return seed
   141		}
   142		keccak256 := makeHasher(sha3.NewLegacyKeccak256())
   143		for i := 0; i < int(block/epochLengthDefault); i++ {
   144			keccak256(seed, seed)
   145		}
   146		return seed
   147	}
```

such as: for block number = 2698351 in mordor

for eth_getWork(), 2698351 // 30000 = 89 times keccak256(seed, seed)
but cache seedhash, (2698521 // 60000) * 60000 + 1 = 2640001, 2640001 // 30000 = 88, only call 88 times.

so, which lead to different seedhash value, thanks!

**Comment by meowsbits** [comment]
@iquidus Is this a dupe of #234?

**Comment by iquidus** [comment]
> @iquidus Is this a dupe of #234?

yes

**Comment by meowsbits** [comment]
Ok, then closing as resolved by #236
